test/com/amazon/ion/BinaryIonReaderIteratorSystemProcessingTest.java

```diff
@@ -0,0 +1,35 @@
+package com.amazon.ion;
+
+import java.util.Iterator;
+
+public class BinaryIonReaderIteratorSystemProcessingTest
+    extends IteratorSystemProcessingTestCase
+{
+    private byte[] myBytes;
+
+    @Override
+    protected void prepare(String text)
+    {
+        myMissingSymbolTokensHaveText = false;
+        myBytes = encode(text);
+    }
+
+    @Override
+    protected Iterator<IonValue> iterate()
+    {
+        return system().iterate(getStreamingMode().newIonReader(system().getCatalog(), myBytes));
+    }
```

Comment:
What does, exactly? The method being called is [`IonSystem.iterate(IonReader)`](https://github.com/amzn/ion-java/blob/master/src/com/amazon/ion/IonSystem.java#L338).

Reply:
The method being called is `IonSystem.iterate(IonReader)`, but conceptually what we're doing is `iterate(bytes)`. I was thinking that `iterate(bytes)` should exist, and it turns out it already does. [`IonSystem.iterate(bytes)`](https://github.com/amzn/ion-java/blob/master/src/com/amazon/ion/IonSystem.java#L320) 🤦🏻‍♂️

In this particular case we can't use `iterate(bytes)` because we need to create differently-configured readers and as discussed elsewhere in this PR we don't have the mutations in IonSystem that we need to make single convenience methods do the right thing for our test cases. I knew that, but somehow still made the jump from "this is awkward to read" to "the 'better' alternative must not exist".

Nothing to do here, confusion about what I meant should be expected.